### PR TITLE
Change OnMessage for new Conversation API.

### PR DESCRIPTION
**Pull request by scottdangelo**
The IBM Watson Conversation service has changed the object
that is returned from a call to the Conversation service.
Formerly, the object was a MessageResponse object that had
fields that could be accessed with dot syntax. Now, the object
is generic and the data returned must be parsed to get the
entities and intents.

### CODE REVIEW

**Review by stevemar — APPROVED** [review]

**Comment by stevemar** [comment]
lgtm, but i'll let other folks have a look too.

File: Creation Sandbox/Assets/Scripts/VoiceSpawner.cs [hunk 1]
```diff
@@ -63,55 +63,78 @@ private void OnSpeechInput(Data data)
         }
     }
 
-    void OnMessage(MessageResponse resp, string customData)
+    void OnMessage(object resp, string customData)
```
Comment (mediumTaj) [inline Creation Sandbox/Assets/Scripts/VoiceSpawner.cs:68]: Something you might want to try in your OnMessage to cast the generic object as a MessageResponse.

```
private void OnMessage(object resp, string data)
    {
        //  Convert resp to fsdata
        fsData fsdata = null;
        fsResult r = _serializer.TrySerialize(resp.GetType(), resp, out fsdata);
        if (!r.Succeeded)
            throw new WatsonException(r.FormattedMessages);

        //  Convert fsdata to MessageResponse
        MessageResponse messageResponse = new MessageResponse();
        object obj = messageResponse;
        r = _serializer.TryDeserialize(fsdata, obj.GetType(), ref obj);
        if (!r.Succeeded)
            throw new WatsonException(r.FormattedMessages);

        Log.Debug("ExampleConversation", "Conversation: Message Response | intent: {0} | confidence : {1}", messageResponse.intents[0].intent, messageResponse.intents[0].confidence);
    }
```

**Review by mediumTaj — CHANGES_REQUESTED** [review]
Check out what I posted to convert the generic object to a MessageResponse.

**Review by mediumTaj — APPROVED** [review]
nice!

**Pull request merged**